test/support/test_helpers.ex

```diff
@@ -683,6 +683,223 @@ defmodule Oli.TestHelpers do
+        content: %{},
+        deleted: false,
+        slug: "unit_container",
+        title: "Unit Container"
+      })
+
+    # Associate unit to the project
+    insert(:project_resource, %{project_id: project.id, resource_id: unit_resource.id})
+
+    # root container
+    container_resource = insert(:resource)
+
+    container_revision =
+      insert(:revision, %{
+        resource: container_resource,
+        objectives: %{},
+        resource_type_id: ResourceType.get_id_by_type("container"),
+        children: [unit_resource.id, page_resource_1.id],
+        content: %{},
+        deleted: false,
+        slug: "root_container",
+        title: "Root Container"
+      })
+
+    # Associate root container to the project
+    insert(:project_resource, %{project_id: project.id, resource_id: container_resource.id})
+
+    # Publication of project with root container
+    publication =
+      insert(:publication, %{
+        project: project,
+        root_resource_id: container_resource.id
+      })
+
+    # Publish root container resource
+    insert(:published_resource, %{
+      publication: publication,
+      resource: container_resource,
+      revision: container_revision,
+      author: author
+    })
+
+    # Publish unit resource
+    insert(:published_resource, %{
+      author: author,
+      publication: publication,
+      resource: unit_resource,
+      revision: unit_revision
+    })
+
+    # Publish module resource
+    insert(:published_resource, %{
+      author: author,
+      publication: publication,
+      resource: module_resource,
+      revision: module_revision
+    })
+
+    # Publish page 1 resource
+    insert(:published_resource, %{
+      author: author,
+      publication: publication,
+      resource: page_resource_1,
+      revision: page_revision_1
+    })
+
+    # Publish page 2 resource
+    insert(:published_resource, %{
+      author: author,
+      publication: publication,
+      resource: page_resource_2,
+      revision: page_revision_2
+    })
+
+    # Publish objective 1 resource
+    insert(:published_resource, %{
+      author: author,
+      publication: publication,
+      resource: obj_resource_1,
+      revision: obj_revision_1
+    })
+
+    # Publish objective 2 resource
+    insert(:published_resource, %{
+      author: author,
+      publication: publication,
+      resource: obj_resource_2,
+      revision: obj_revision_2
+    })
+
+    section =
+      insert(:section,
+        base_project: project,
+        context_id: UUID.uuid4(),
+        open_and_free: true,
+        registration_open: true,
+        type: :enrollable
+      )
+
+    {:ok, section} = Sections.create_section_resources(section, publication)
+    Sections.rebuild_contained_pages(section)
+
+    [
+      project: project,
+      section: section,
+      publication: publication,
+      obj_revision_1: obj_revision_1,
+      obj_revision_2: obj_revision_2,
+      module_revision: module_revision
+    ]
```

Comment:
What do you think about returning a map? With a map we can pattern match against a single key-value when we invoke this function inside a test (not necessarily in the setup), since we might not need all the fields in some cases.